google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberBuilder.java

```diff
@@ -77,19 +89,17 @@ public Subscriber build() throws ApiException {
         serviceClient = autoBuilt.serviceClient().get();
       } else {
         try {
-          Map<String, String> metadata = autoBuilt.context().getMetadata();
-          Map<String, String> routingMetadata =
-              RoutingMetadata.of(autoBuilt.subscriptionPath(), autoBuilt.partition());
-          Map<String, String> allMetadata =
-              ImmutableMap.<String, String>builder()
-                  .putAll(metadata)
-                  .putAll(routingMetadata)
-                  .build();
+          SubscriberServiceSettings.Builder settingsBuilder =
+              SubscriberServiceSettings.newBuilder();
+          addDefaultMetadata(
+              autoBuilt.context(),
+              autoBuilt.subscriptionPath(),
+              autoBuilt.partition(),
+              SubscriberServiceSettings.newBuilder());
```

Comment:
This should be settingsBuilder: As of now, it modifies an object then discards it.